semantic/utilities.go

```diff
@@ -0,0 +1,24 @@
+package semantic
+
+import (
+	"fmt"
+	"math/big"
+)
+
+func convertToBigInt(str string) (*big.Int, error, bool) {
```

Comment:
for now I went with having this return a `bool` along with an `error` because a lot of the existing code was `v, vIsNumber` and it felt weird to either rename it to `vErr` or to keep it the name but compare it against `nil`.

I don't think this is super bad, but now that I've got everything transitioned if others felt it would be more Go-y to change all the checks to be checking the `err`, I can do that

Reply:
Here's a concrete example of what I mean:

```
vv, vErr := convertToBigInt(vt.value)
wv, wErr := convertToBigInt(wt.value)

// numeric tokens have the same natural order
if vErr != nil && wErr != nil {
	return vv.Cmp(wv) == -1, nil
}
```

vs

```
vv, vIsNumber := convertToBigInt(vt.value)
wv, wIsNumber := convertToBigInt(wt.value)

// numeric tokens have the same natural order
if vIsNumber != nil && wIsNumber != nil {
	return vv.Cmp(wv) == -1, nil
}
```

though, I've just realised I could probably actually switch to checking if `vv` is `nil`...? but I think that has a similar trade off in readability 🤔

Reply:
I can see the benefit from introducing a boolean for isNumber, but we should make sure people understand what the code is supposed to return. Either add a comment about the return values or make them named, e.g.
```
func convertToBigInt(str string) (res *big.Int, err error, isNumber bool) {
```